client/config_client.go

```diff
@@ -0,0 +1,191 @@
+// Copyright 2019 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pd
+
+import (
+	"context"
+	"time"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/pingcap/kvproto/pkg/configpb"
+	"github.com/pingcap/log"
+	"github.com/pkg/errors"
+	"go.uber.org/zap"
+)
+
+// ConfigClient is a client to manage the configuration.
+// It should not be used after calling Close().
+type ConfigClient interface {
+	GetClusterID(ctx context.Context) uint64
+	Create(ctx context.Context, v *configpb.Version, component, componentID, config string) (*configpb.Status, *configpb.Version, string, error)
+	Get(ctx context.Context, v *configpb.Version, component, componentID string) (*configpb.Status, *configpb.Version, string, error)
+	Update(ctx context.Context, v *configpb.Version, kind *configpb.ConfigKind, entries []*configpb.ConfigEntry) (*configpb.Status, *configpb.Version, error)
+	Delete(ctx context.Context, v *configpb.Version, kind *configpb.ConfigKind) (*configpb.Status, error)
+	// Close closes the client.
+	Close()
+}
+
+type configClient struct {
+	*baseClient
+}
+
+// NewConfigClient creates a PD configuration client.
+func NewConfigClient(pdAddrs []string, security SecurityOption) (ConfigClient, error) {
+	return NewConfigClientWithContext(context.Background(), pdAddrs, security)
+}
+
+// NewConfigClientWithContext creates a PD configuration client with the context.
+func NewConfigClientWithContext(ctx context.Context, pdAddrs []string, security SecurityOption) (ConfigClient, error) {
+	log.Info("[pd] create pd configuration client with endpoints", zap.Strings("pd-address", pdAddrs))
+	base, err := newBaseClient(ctx, addrsToUrls(pdAddrs), security)
+	if err != nil {
+		return nil, err
+	}
+	return &configClient{base}, nil
+}
+
+func (c *configClient) Close() {
```

Comment:
Almost identical to `pd.Client.Close`, how about moving it `baseClient`?

Reply:
pd client will call `revokeTSORequest` when closing it.